osidb/serializer.py

```diff
@@ -505,6 +507,54 @@ class Meta:
         abstract = True
 
 
+class JiraTaskSyncMixinSerializer(serializers.ModelSerializer):
+    """
+    serializer mixin class implementing special handling of the models
+    which need to perform Jira sync as part of the save procedure
+    """
+
+    def __get_user_jira_token(self):
+        jira_token = self.context["request"].META.get("HTTP_JIRA_API_KEY")
+        if not jira_token:
+            raise serializers.ValidationError(
+                {"Jira-Api-Key": "This HTTP header is required."}
+            )
+        return jira_token
+
+    def create(self, validated_data):
+        """
+        perform the ordinary instance create
+        with providing Jira token while saving
+        """
+        flaw = super().create(validated_data)
+        if JIRA_TASKMAN_AUTO_SYNC_FLAW:
+            flaw.tasksync(
+                jira_token=self.__get_user_jira_token(), creation_required=True
+            )
+        return flaw
```

Comment:
Furthermore you should instead do like the `BugzillaSyncMixinSerializer` which creates an instance object and then saves it passing the jira token as an argument to `save()`, since the `save()` override will check for the `JIRA_TASKMAN_AUTO_SYNC_FLAW` feature flag.